pom.xml

```diff
@@ -85,6 +85,16 @@
       <artifactId>credentials</artifactId>
       <version>2.1.5</version>
     </dependency>
+    <dependency>
+      <groupId>org.jenkins-ci</groupId>
+      <artifactId>symbol-annotation</artifactId>
+      <version>1.3</version>
+    </dependency>
+    <dependency>
+      <groupId>org.jenkins-ci</groupId>
+      <artifactId>annotation-indexer</artifactId>
```

Comment:
Why do we need annotation-indexer?

Reply:
by only adding the dependency on `org.jenkins-ci:symbol-annotation`

I get following build error:

> [INFO] --- maven-enforcer-plugin:3.0.0-M1:enforce (display-info) @ stash-pullrequest-builder ---
[INFO] Ignoring requireUpperBoundDeps in com.google.guava:guava
[INFO] Ignoring requireUpperBoundDeps in com.google.code.findbugs:jsr305
[WARNING] Rule 4: org.apache.maven.plugins.enforcer.RequireUpperBoundDeps failed with message:
Failed while enforcing RequireUpperBoundDeps. The error(s) are [
Require upper bound dependencies error for org.jenkins-ci:annotation-indexer:1.9 paths to dependency are:
+-org.jenkins-ci.plugins:stash-pullrequest-builder:1.9-SNAPSHOT
  +-org.jenkins-ci:symbol-annotation:1.3
    +-org.jenkins-ci:annotation-indexer:1.9
and
+-org.jenkins-ci.plugins:stash-pullrequest-builder:1.9-SNAPSHOT
  +-org.jenkins-ci.main:jenkins-core:2.60.1
    +-org.jenkins-ci:annotation-indexer:1.11
and
+-org.jenkins-ci.plugins:stash-pullrequest-builder:1.9-SNAPSHOT
  +-org.kohsuke:access-modifier-annotation:1.14
    +-org.jenkins-ci:annotation-indexer:1.4
and
+-org.jenkins-ci.plugins:stash-pullrequest-builder:1.9-SNAPSHOT
  +-org.jenkins-ci.plugins:git:3.0.0
    +-com.infradna.tool:bridge-method-annotation:1.14
      +-org.jenkins-ci:annotation-indexer:1.4
and
+-org.jenkins-ci.plugins:stash-pullrequest-builder:1.9-SNAPSHOT
  +-org.jenkins-ci.main:jenkins-core:2.60.1
    +-org.jenkins-ci:bytecode-compatibility-transformer:1.8
      +-org.jenkins-ci:annotation-indexer:1.4
]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------

Reply:
OK. The way to solve this is to lock the version using `dependencyManagement`, not to add a new direct dependency.
https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#Dependency_Management

Reply:
thx for pointing that out. Moved to `dependencyManagement`